src/lib.rs

```diff
@@ -25,12 +25,38 @@ use defmt::Format;
 use embedded_hal::digital::OutputPin;
 use embedded_hal::pwm::SetDutyCycle;
 
-/// Defines errors which can happen while trying to set a speed.
+/// Defines errors which can happen when calling [`Motor::drive()`].
 #[derive(PartialEq, Eq, Debug, Copy, Clone)]
 #[cfg_attr(feature = "defmt", derive(Format))]
-pub enum DriveError {
+pub enum DriveError<IN1Error, IN2Error, PWMError> {
     /// An invalid speed has been defined. The speed must be given as a percentage value between 0 and 100 to be valid.
     InvalidSpeed,
+    /// An error in setting the output of the IN1 pin
+    In1Error(IN1Error),
+    /// An error in setting the output of the IN2 pin
+    In2Error(IN2Error),
+    /// An error in setting the output of the PWM pin
+    PwmError(PWMError),
+}
+
+/// Defines errors which can happen when calling [`Tb6612fng::new()`].
+#[derive(PartialEq, Eq, Debug, Copy, Clone)]
+#[cfg_attr(feature = "defmt", derive(Format))]
+pub enum InitError<
```

Comment:
how about
```suggestion
pub enum Tb6612fngError<
```
?

and renaming `DriveError` to `MotorError`? then we have one error-enum per struct. this _might_ make things clearer?

Reply:
I think that's a good idea, consistency like this is always nice.